### DISCUSSION TITLE
ValueError: Digitalized labels does not have elements close enough to bin index 1. The bin index should be in the range of the labels values.

### DISCUSSION THREAD

**Discussion by FrancescaGuo** [AdaptiveMotorControlLab/CEBRA#106] [Q&A]
I have been struggling with this value error when I use the Consistency demo codes on my own mice behavioral data. Here are a few screenshots for my codes and a full description of the error. I have trained the Cebra_Behavior model with mice neural data and behavioral data. I successfully get the embeddings and visualize them but get stuck when I try to calculate the consistency of the embeddings between mice. 
![Screenshot 2023-11-13 182917](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/59901749/b6ab983d-bbbd-4128-bb09-5f72a53d22e0)
'Mouselabel' is a 2D list which has the same shape as my embeddings. 'Names' is a list of integers with the mouse ID (1 to 14).
![Screenshot 2023-11-13 183015](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/59901749/93c81530-5dea-4e9f-8ec6-90c0fdb26597)
![Screenshot 2023-11-13 182950](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/59901749/6ea0d9fe-c7b8-44da-8549-b6dd2e2acfdd)
I would really appreciate it if someone has any clue of what I did wrong.

**Comment by stes** [comment]
Hi @FrancescaGuo , apologies for the late response here. Please also see my related reply here: https://github.com/AdaptiveMotorControlLab/CEBRA/discussions/125#discussioncomment-8447363

The code uses the default number of bins (100) and partitions the behavioral data into this number of bins. In your example, it would be a reasonable choice to set the number of bins to the number of discrete labels (14).

**Comment by stes** [comment]
I created an issue to improve the code here: https://github.com/AdaptiveMotorControlLab/CEBRA/issues/128

In the meantime, please consider setting `n_bins` manually to the number of discrete labels. Let me know if this works!

**Comment by stes** [comment]
Hello @FrancescaGuo , just checking, were you able to fix the issue by adapting the bin size?

**Reply by ELGarulli** [reply]
Hi @stes I was getting the same error as @FrancescaGuo and setting the num_discretization_bins to the unique values in my labels solved it. 
I just want to mention that at first I was confused when you suggested "number of discrete labels (14)", as I understand OP has 14 single session models to compare, but the number of values of labels could be any other number, is this correct?
I am adding this note as I would like to check if I misunderstood something. I am trying to train a multi-session model. Each session is a separate run (s=55), my labels are a list of arrays each as long as the corresponding run, each label array can contain max 3 values (0, 1, 2).
So for me setting the num_discretization_bins to 3 is what solves the "Digitalized labels does not have elements close enough to bin" error.
